docs/sources/guides/dataframe.md

```diff
@@ -0,0 +1,157 @@
++++
+title = "Introduction Dataframes"
+description = "Guide to understanding Dataframes"
+keywords = ["grafana", "dataframe", "guide"]
+type = "docs"
+aliases = ["/docs/grafana/latest/guides/dataframe"]
+[menu.docs]
+name = "Dataframes"
+identifier = "dataframes_guide"
+parent = "guides"
+weight = 350
++++
+
+## Introduction to Dataframes
+
+Dataframes are a new concept in Grafana 7.0 and they replace the "Time Series" and "Table" types in Grafana. Dataframes are a more generic structure that can hold different shapes of time series, tables, and other types.
+
+A dataframe is a [Columnar oriented](https://en.wikipedia.org/wiki/Column-oriented_DBMS) table like structure, meaning it stores data by column and not by row.
+
+A simplified conceptual model of a dataframe is:
+
+```ts
+interface DataFrame {
+    name?:  string;
+    // reference to query that create the frame
+    refId?: string;
+
+    fields: []Field;
+}
+```
+
+```ts
+interface Field {
+    name:    string;
+     // Prometheus like Labels / Tags
+    labels?: Record<string, string>;
+
+    // For example string, number, time (or more specific primitives in the backend)
+    type:   FieldType;
+    // Array of values all of the same type
+    values: Vector<T>;
+
+    // Optional display data for the field (e.g. unit, name over-ride, etc)
+    config: FieldConfig;
+}
+```
+
+With dataframes, each column is represented by a **Field**. So the essence of a dataframe is an Field array with additional properties. Those additional properties on a Field include Name, Labels (a.k.a (Tags)), Optional display data and the data type of the Field's values. There are also additional properties display and metadata properties on the Frame itself.
+
+One restriction on dataframes is that all Fields in the frame must be of the same length to be a valid dataframe.
+
+### When is a Dataframe a Time Series or a Table
```

Comment:
Per the [Documentation style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md), headings should be sentence case.